server/src/main/java/org/opensearch/gateway/ClusterStateUpdaters.java

```diff
@@ -120,7 +121,19 @@ static ClusterState updateRoutingTable(final ClusterState state) {
         // initialize all index routing tables as empty
         final RoutingTable.Builder routingTableBuilder = RoutingTable.builder(state.routingTable());
         for (final IndexMetadata cursor : state.metadata().indices().values()) {
-            routingTableBuilder.addAsRecovery(cursor);
+            // Whether IndexMetadata is recovered from local disk or remote it doesn't matter to us at this point.
+            // We are only concerted about index data recovery here. Which is why we only check for remote store enabled and not for remote
+            // cluster state enabled.
+            if (cursor.getSettings().getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false) == false
+                || state.routingTable()
+                    .shardsMatchingPredicateCount(
+                        shardRouting -> shardRouting.primary()
+                            // We need to ensure atleast one of the primaries is being recovered from remote.
+                            // This ensures we have gone through the RemoteStoreRestoreService and routing table is updated
+                            && shardRouting.recoverySource() instanceof RecoverySource.RemoteStoreRecoverySource
+                    ) == 0) {
```

Comment:
Ideally, this count should either be 0 or number of shards. If it is neither, we will have red index as some of the shards will be unassigned. But given that this is remote backed index, we should be able to recover it manually using `_remotestore/_restore` API. Is this understanding correct?

Reply:
yes, if IndexMetadata is present in the cluster you can always call the _remotestore/_restore api to make it green